addons/stock/models/stock_inventory.py

```diff
@@ -200,13 +200,54 @@ def action_cancel_draft(self):
     @api.multi
     def action_start(self):
         for inventory in self:
-            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             if (inventory.filter != 'partial') and not inventory.line_ids:
-                vals.update({'line_ids': [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]})
+                line_values_list = inventory._get_inventory_lines_values()
+                line_ids = []
+                for line_value in line_values_list:
+                    (sql, params) = self.inventory_line_to_sql_insert(line_value, inventory)
+                    self.env.cr.execute(sql, params)
+                    line_ids.append(self._cr.fetchone()[0])
+                lines = self.env['stock.inventory.line'].browse(line_ids)
+                lines.modified(self.env['stock.inventory.line']._fields)
+
+            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             inventory.write(vals)
+        self.recompute()
         return True
+
     prepare_inventory = action_start
 
+    def inventory_line_to_sql_insert(self, iline, inventory):
```

Comment:
could put this methode private ?
add check_acces_right() to check right to make the SQL query.

cc @odony